gymnasium/experimental/wrappers/jax_to_torch.py

```diff
@@ -1,178 +1,178 @@
-
-        Args:
-            seed: The seed for resetting the environment
-            options: The options for resetting the environment, these are converted to jax arrays.
-
-        Returns:
-            PyTorch-based observations and info
-        """
-        if options:
-            options = torch_to_jax(options)
-
-        return jax_to_torch(self.env.reset(seed=seed, options=options), self.device)
-
-    def render(self) -> RenderFrame | list[RenderFrame] | None:
-        """Returns the rendered frames as a NumPy array."""
-        return jax_to_numpy(self.env.render())
+# This wrapper will convert torch inputs for the actions and observations to Jax arrays
+# for an underlying Jax environment then convert the return observations from Jax arrays
+# back to torch tensors.
+#
+# Functionality for converting between torch and jax types originally copied from
+# https://github.com/google/brax/blob/9d6b7ced2a13da0d074b5e9fbd3aad8311e26997/brax/io/torch.py
+# Under the Apache 2.0 license. Copyright is held by the authors
+
+"""Helper functions and wrapper class for converting between PyTorch and Jax."""
+from __future__ import annotations
+
+import functools
+import numbers
+from collections import abc
+from typing import Any, Iterable, Mapping, SupportsFloat, Union
+
+import gymnasium as gym
+from gymnasium.core import RenderFrame, WrapperActType, WrapperObsType
+from gymnasium.error import DependencyNotInstalled
+from gymnasium.experimental.wrappers.jax_to_numpy import jax_to_numpy
+
+
+try:
+    import jax.numpy as jnp
+    from jax import dlpack as jax_dlpack
+except ImportError:
+    raise DependencyNotInstalled(
+        "Jax is not installed therefore cannot call `torch_to_jax`, run `pip install gymnasium[jax]`"
+    )
+
+try:
+    import torch
+    from torch.utils import dlpack as torch_dlpack
+
+    Device = Union[str, torch.device]
+except ImportError:
+    raise DependencyNotInstalled(
+        "Torch is not installed therefore cannot call `torch_to_jax`, run `pip install torch`"
+    )
+
+
+__all__ = ["jax_to_torch", "torch_to_jax", "JaxToTorchV0"]
+
+
+@functools.singledispatch
+def torch_to_jax(value: Any) -> Any:
+    """Converts a PyTorch Tensor into a Jax DeviceArray."""
+    raise Exception(
+        f"No known conversion for Torch type ({type(value)}) to Jax registered. Report as issue on github."
+    )
+
+
+@torch_to_jax.register(numbers.Number)
+def _number_torch_to_jax(value: numbers.Number) -> Any:
+    """Convert a python number (int, float, complex) to a jax array."""
+    return jnp.array(value)
+
+
+@torch_to_jax.register(torch.Tensor)
+def _tensor_torch_to_jax(value: torch.Tensor) -> jnp.DeviceArray:
+    """Converts a PyTorch Tensor into a Jax DeviceArray."""
+    tensor = torch_dlpack.to_dlpack(value)  # pyright: ignore[reportPrivateImportUsage]
+    tensor = jax_dlpack.from_dlpack(tensor)  # pyright: ignore[reportPrivateImportUsage]

```

Comment:
Example of the same pyright comment being done for the other usage of `from_dlpack`: `# pyright: ignore[reportPrivateImportUsage]`